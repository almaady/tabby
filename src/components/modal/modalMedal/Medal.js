import React from "react";
import PropTypes from "prop-types";
/**
 * @Class Medal - The image of the medal
 */
class Medal extends React.PureComponent {
  /**
   * @property {string} medalSrc - Medal URL
   */
  static propTypes = {
    medalSrc: PropTypes.string.isRequired
  };

  render() {
    const { medalSrc } = this.props;
    return <img alt="Medal" src={medalSrc} />;
  }
}

export default Medal;
