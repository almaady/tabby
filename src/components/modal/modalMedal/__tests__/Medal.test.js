import React from "react";
import { shallow } from "enzyme";

import Medal from "../Medal";

describe("Medal", () => {
  let props;
  beforeEach(() => {
    props = {
      medalSrc: "URL"
    };
  });
  it("should render correctly with default state", () => {
    const component = shallow(<Medal {...props} />);
    expect(component).toMatchSnapshot();
  });
});
