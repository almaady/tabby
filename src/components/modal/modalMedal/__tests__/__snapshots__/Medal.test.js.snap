// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Medal should render correctly with default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Medal
    medalSrc="URL"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "Medal",
      "src": "URL",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Medal",
        "src": "URL",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
